syntax = "proto3";

service DataNode {
  rpc WriteBlock (BlockRequest) returns (StatusResponse);
  rpc ReadBlock (BlockRequest) returns (BlockResponse);
}

message BlockRequest {
  string block_id = 1;
  bytes data = 2;
}

message BlockResponse {
  bytes data = 1;
}

message StatusResponse {
  bool success = 1;
}